### ISSUE TITLE
Issue with 2d coordinate accuracy

### ISSUE THREAD

**Issue by LucasHaberkamp** [CMU-Perceptual-Computing-Lab/openpose#1820]
### Posting rules
1. **Duplicated posts will not be answered**. Check the [FAQ](https://github.com/CMU-Perceptual-Computing-Lab/openpose/blob/master/doc/faq.md) section, other GitHub issues, and general documentation before posting. E.g., **low-speed, out-of-memory, output format, 0-people detected, installation issues, ...**).
2. **Fill** the **Your System Configuration section (all of it or it will not be answered!)** if you are facing an error or unexpected behavior. Feature requests or some other type of posts might not require it.
3. **No questions about training or 3rd party libraries**:
    - OpenPose only implements testing.
    - Caffe errors/issues, check [Caffe](http://caffe.berkeleyvision.org) documentation.
    - CUDA check failed errors: They are usually fixed by re-installing CUDA, then re-installing the proper cuDNN version, and then re-compiling (or re-installing) OpenPose. Otherwise, check for help in CUDA forums.
    - OpenCV errors: Install the default/pre-compiled OpenCV or check for online help.
4. Set a **proper issue title**: add the Ubuntu/Windows keyword and be specific (e.g., do not call it: `Error`).
5. Only English comments.
Posts which do not follow these rules will be **ignored, closed, or reported** with no further clarification.

### Issue Summary
When analyzing the 2d coordinates extracted from the JSON file, the coordinates are not as expected and don't match up with what is displayed on the video. As when I go frame by frame following a specific point, the extracted coordinates do not correspond to an expected output. I.e. as an individual runs across the screen the x coordinate does not increase from the start of the video. Potentially, I am using the wrong flag when writing the JSON results, however I can't figure out where the mistake is. 

### Executed Command (if any)
Note: add `--logging_level 0 --disable_multi_thread` to get higher debug information.

!cd openpose && ./build/examples/openpose/openpose.bin --video ../video.mp4 --part_candidates --number_people_max 1 --write_json ./output/ --display 0 --write_video ../openpose.avi

### OpenPose Output (if any)
Below is a youtube video link which pertains to the open pose coordinates.

https://youtu.be/oeTA_qIewS8

Also, below is the coordinates of the right knee of the above youtube video.

  | x | y | acc
-- | -- | -- | --
0 | 954 | 368 | 0
1 | 833 | 365 | 0
2 | 530 | 433 | 0
3 | 1539 | 362 | 0
4 | 686 | 388 | 0
5 | 530 | 433 | 0
6 | 880 | 409 | 0
7 | 880 | 386 | 0
8 | 880 | 412 | 0
9 | 530 | 433 | 0
10 | 533 | 430 | 0
11 | 1286 | 318 | 0
12 | 1092 | 321 | 0
13 | 868 | 430 | 0
14 | 912 | 389 | 0
15 | 530 | 433 | 0
16 | 1454 | 386 | 0
17 | 530 | 433 | 0
18 | 530 | 433 | 0
19 | 556 | 412 | 0
20 | 789 | 365 | 0
21 | 871 | 430 | 0
22 | 530 | 433 | 0
23 | 868 | 427 | 0
24 | 877 | 421 | 0
25 | 530 | 433 | 0
26 | 1610 | 335 | 0
27 | 1469 | 409 | 0
28 | 1351 | 338 | 0
29 | 1018 | 341 | 0
30 | 530 | 433 | 0
31 | 1469 | 412 | 0
32 | 530 | 433 | 0
33 | 1466 | 418 | 0
34 | 595 | 403 | 0
35 | 1422 | 347 | 0
36 | 530 | 433 | 0
37 | 860 | 371 | 0
38 | 530 | 433 | 0
39 | 530 | 433 | 0
40 | 530 | 433 | 0
41 | 624 | 418 | 0
42 | 0 | 0 | 0
43 | 544 | 427 | 0
44 | 742 | 365 | 0
45 | 1492 | 389 | 0
46 | 1469 | 415 | 0

### Type of Issue
You might select multiple topics, delete the rest:
- Compilation/installation error
- Execution error
- Help wanted
- Question
- Enhancement / offering possible extensions / pull request / etc
- Other (type your own type)

### Your System Configuration
1. **Whole console output** (if errors appeared), paste the error to [PasteBin](https://pastebin.com/) and then paste the link here: LINK

2. **OpenPose version**: Latest GitHub code? Or specific commit (e.g., d52878f)? Or specific version from `Release` section (e.g., 1.2.0)?

3. **General configuration**:
    - **Installation mode**: CMake, sh script, manual Makefile installation, ... (Ubuntu); CMake, ... (Windows); ...?
    - **Operating system** (`lsb_release -a` in Ubuntu):
    - **Operating system version** (e.g., Ubuntu 16, Windows 10, ...):
    - **Release or Debug mode**? (by default: release):
    - Compiler (`gcc --version` in Ubuntu or VS version in Windows): 5.4.0, ... (Ubuntu); VS2015 Enterprise Update 3, VS2017 community, ... (Windows); ...?

4. **Non-default settings**:
    - **3-D Reconstruction module added**? (by default: no):
    - Any other custom CMake configuration with respect to the default version? (by default: no):

5. **3rd-party software**:
    - **Caffe version**: Default from OpenPose, custom version, ...?
    - **CMake version** (`cmake --version` in Ubuntu):
    - **OpenCV version**: pre-compiled `apt-get install libopencv-dev` (only Ubuntu); OpenPose default (only Windows); compiled from source? If so, 2.4.9, 2.4.12, 3.1, 3.2?; ...?

6. If **GPU mode** issue:
    - **CUDA version** (`cat /usr/local/cuda/version.txt` in most cases):
    - **cuDNN version**:
    - **GPU model** (`nvidia-smi` in Ubuntu):

7. If **CPU-only mode** issue:
    - **CPU brand & model**:
    - Total **RAM memory** available:

8. If **Python** API:
    - **Python version**: 2.7, 3.7, ...?
    - **Numpy version** (`python -c "import numpy; print numpy.version.version"` in Ubuntu):

9. If **Windows** system:
    - Portable demo or compiled library?

10. If **speed performance** issue:
    - Report OpenPose timing speed based on [this link](https://github.com/CMU-Perceptual-Computing-Lab/openpose/blob/master/doc/speed_up_openpose.md#profiling-speed).

**Comment by gineshidalgo99** [comment]
3 ideas here, let me know in your case!

1. Just to check, but I don't think it is this: Does your origin matches OpenPose? Ie OpenPose (and OpenCV) assumes that the X coordinate goes from 0 (left) to width-1 (right) and the Y coordinate from 0 (top) to height-1 (bottom)
2. My bet: Why are you using part_candidates? You should not use part_candidates here, you should just be checking the knee position from the skeleton, not from the part_candidates field. 
3. If it's none of the above, could you re-run that video (https://youtu.be/oeTA_qIewS8) without --number_people_max 1? I am concerned the issue might be > 1 person.

Let me know if any of this fixes it! :)

**Comment by LucasHaberkamp** [comment]
Thanks for you help, the second idea was the fix! Once I removed the part_candidates field I was able to access the data I wanted in the JSON file. Also, I had a for loop to access the JSON files as shown below.

for file in sorted(json_files):
    x = json.load(open(path_to_json + file))

The files were out of order until I added sorted. This allowed me to extract the coordinates in the correct order.
